frontend/src/components/Summary/ImplementationRow.jsx

```diff
@@ -1,24 +1,22 @@
 import "./ImplementationRow.css";
-import { InfoCircleFill } from "react-bootstrap-icons";
 import { useState } from "react";
 import { DetailsButtonModal } from "../Modals/DetailsButtonModal";
-import { RunTimeInfoModal } from "../Modals/RunTimeInfoModal";
 import { mapLanguage } from "../../data/mapLanguage";
+import { NavLink } from "react-router-dom";
 
 const ImplementationRow = ({ cases, implementation }) => {
   const [showDetails, setShowDetails] = useState(false);
-  const [showRunTimeInfo, setShowRunTimeInfo] = useState(false);
 
+  const getPath = () => {
+    const pathSegment = implementation.id.split("/");
```

Comment:
Can we encapsulate this? Perhaps with a property / function on `implementation`? What type of object is that? Can we teach it to know what this value is?

Reply:
I've now converted this file to `tsx` in order to add **Type** annotations.

Reply:
That wasn't precisely what I meant, though seems like a good idea anyhow. What I meant is "can we put this function on a class or javascript object which represents an `Implementation`"? It seems likely to be an operation we may want to do in a bunch of places in the code base. Specifically I think someone should want to have a JavaScript accessor which lets them write `implementation.imageName` (or some appropriate name) which returns this value, but that doesn't involve the caller/user knowing how to do this parsing.